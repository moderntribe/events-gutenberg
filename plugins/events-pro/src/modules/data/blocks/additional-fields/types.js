/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_PRO_STORE } from '@moderntribe/events-pro/data/prefix';

export const ADD_ADDITIONAL_FIELD = `${ PREFIX_EVENTS_PRO_STORE }/ADD_ADDITIONAL_FIELD`;
export const REMOVE_ADDITIONAL_FIELD = `${ PREFIX_EVENTS_PRO_STORE }/REMOVE_ADDITIONAL_FIELD`;

export const SET_ADDITIONAL_FIELD_LABEL = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_LABEL`;
export const SET_ADDITIONAL_FIELD_IS_PRISTINE = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_IS_PRISTINE`;
export const SET_ADDITIONAL_FIELD_VALUE = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_VALUE`;
export const SET_ADDITIONAL_FIELD_TYPE = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_TYPE`;
export const SET_ADDITIONAL_FIELD_OPTIONS = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_OPTIONS`;
export const SET_ADDITIONAL_FIELD_DIVIDER_LIST = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_DIVIDER_LIST`;
export const SET_ADDITIONAL_FIELD_DIVIDER_END = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_DIVIDER_END`;
