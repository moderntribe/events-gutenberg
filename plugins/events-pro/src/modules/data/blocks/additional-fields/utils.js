export const FIELD_TYPES = {
	CHECKBOX: 'CHECKBOX',
	TEXT: 'TEXT',
	DROP_DOWN: 'DROP_DOWN',
	URL: 'URL',
	RADIO: 'RADIO',
	TEXT_AREA: 'TEXT_AREA',
};
