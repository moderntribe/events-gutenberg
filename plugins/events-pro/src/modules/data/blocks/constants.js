//
// ─── BLOCK TYPES ────────────────────────────────────────────────────────────────
//


export const RECURRING = 'recurring';
export const EXCEPTION = 'exception';

export const BLOCK_TYPES = [ RECURRING, EXCEPTION ];

//
// ─── STATE KEYS ─────────────────────────────────────────────────────────────────
//

export const KEY_TYPE = 'type';
export const KEY_ALL_DAY = 'all_day';
export const KEY_MULTI_DAY = 'multi_day';
export const KEY_START_TIME = 'start_time';
export const KEY_END_TIME = 'end_time';
export const KEY_START_DATE = 'start_date';
export const KEY_LIMIT = 'limit';
export const KEY_LIMIT_TYPE = 'limit_type';
export const KEY_BETWEEN = 'between';
export const KEY_TIMEZONE = 'timezone';
