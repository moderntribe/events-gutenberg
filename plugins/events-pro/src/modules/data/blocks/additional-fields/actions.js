/**
 * Internal dependencies
 */
import { types } from '@moderntribe/events-pro/data/blocks/additional-fields';

export const addField = ( name ) => ( {
	type: types.ADD_ADDITIONAL_FIELD,
	payload: {
		name,
	},
} );

export const removeField = ( name ) => ( {
	type: types.REMOVE_ADDITIONAL_FIELD,
	payload: {
		name,
	},
} );

export const setFieldLabel = ( name, label ) => ( {
	type: types.SET_ADDITIONAL_FIELD_LABEL,
	payload: {
		name,
		label,
	},
} );

export const setFieldIsPristine = ( name, isPristine ) => ( {
	type: types.SET_ADDITIONAL_FIELD_IS_PRISTINE,
	payload: {
		name,
		isPristine,
	},
} );

export const setFieldValue = ( name, value ) => ( {
	type: types.SET_ADDITIONAL_FIELD_VALUE,
	payload: {
		name,
		value,
	},
} );

export const setFieldType = ( name, type ) => ( {
	type: types.SET_ADDITIONAL_FIELD_TYPE,
	payload: {
		name,
		type,
	},
} );

export const setFieldOptions = ( name, options ) => ( {
	type: types.SET_ADDITIONAL_FIELD_OPTIONS,
	payload: {
		name,
		options,
	},
} );

export const setFieldDividerList = ( name, dividerList ) => ( {
	type: types.SET_ADDITIONAL_FIELD_DIVIDER_LIST,
	payload: {
		name,
		dividerList,
	},
} );

export const setFieldDividerEnd = ( name, dividerEnd ) => ( {
	type: types.SET_ADDITIONAL_FIELD_DIVIDER_END,
	payload: {
		name,
		dividerEnd,
	},
} );

