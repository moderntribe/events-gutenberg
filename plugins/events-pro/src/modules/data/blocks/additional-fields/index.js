/**
 * Internal dependencies
 */
import reducer from './reducers';
import * as types from './types';
import * as actions from './actions';
import * as selectors from './selectors';
import * as utils from './utils';

export default reducer;
export { types, actions, selectors, utils };
