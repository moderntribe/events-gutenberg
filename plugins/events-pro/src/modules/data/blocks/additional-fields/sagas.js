/**
 * External dependencies
 */
import { takeEvery, put, select, all } from 'redux-saga/effects';
import noop from 'lodash/noop';
import isEmpty from 'lodash/isEmpty';
import identity from 'lodash/identity';

/**
 * Internal dependencies
 */
import { actions, types, selectors } from '@moderntribe/events-pro/data/blocks/additional-fields';
import { FIELD_TYPES } from '@moderntribe/events-pro/blocks/additional-fields/utils';

export function* setInitialState( props ) {
	const { payload = {} } = props;
	const { name = '', get = noop } = payload;
	const type = get( 'type' );
	const value = get( 'value' );

	yield all( [
		put( actions.addField( name ) ),
		put( actions.setFieldIsPristine( name, get( 'isPristine' ) ) ),
		put( actions.setFieldType( name, type ) ),
		put( actions.setFieldLabel( name, get( 'label' ) ) ),
		put( actions.setFieldOptions( name, get( 'options' ) ) ),
		put( actions.setFieldValue( name, get( 'value' ) ) ),
	] );
}

export function* setPristineState( props ) {
	const { payload = {} } = props;
	const { name = '' } = payload;
	const value  = yield select( selectors.getFieldValue, payload );
	const isPristine = value === null || '' === value || isEmpty( value );
	yield put( actions.setFieldIsPristine( name, isPristine ) );
}

export function* appendFieldValue( props ) {
	const { payload = {} } = props;
	const { name = '', value } = payload;
	const current = yield select( selectors.getFieldCheckboxValue, payload );
	const newValue = [ ...current, value ]
		.filter( identity )
		.join( '|' );
	yield put( actions.setFieldValue( name, newValue ) );
}

export function* removeFieldValue( props ) {
	const { payload = {} } = props;
	const { name = '', value } = payload;
	const current = yield select( selectors.getFieldCheckboxValue, payload );
	const newValue = current
		.filter( ( text ) => text !== value )
		.join( '|' );
	yield put( actions.setFieldValue( name, newValue ) );
}


export default function* watchers() {
	yield takeEvery( types.SET_ADDITIONAL_FIELD_INITIAL_STATE, setInitialState );
	yield takeEvery( types.SET_ADDITIONAL_FIELD_CHANGE, setPristineState );
	yield takeEvery( types.APPEND_ADDITIONAL_FIELD_VALUE, appendFieldValue );
	yield takeEvery( types.REMOVE_ADDITIONAL_FIELD_VALUE, removeFieldValue );
}
