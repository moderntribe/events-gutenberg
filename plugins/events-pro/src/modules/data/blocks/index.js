/**
 * Internal dependencies
 */
import * as constants from './constants';
import reducer from './reducer';
import './sagas';

export default reducer;
export { constants };
