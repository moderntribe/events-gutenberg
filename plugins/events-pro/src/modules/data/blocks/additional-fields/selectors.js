/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { constants } from '@moderntribe/common/data/plugins';

export const getPlugin = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ];
export const getBlocks = createSelector(
	[ getPlugin ],
	( plugin ) => plugin.blocks,
);
export const getAdditionalFields = createSelector(
	[ getBlocks ],
	( blocks ) => blocks.additionalFields,
);
