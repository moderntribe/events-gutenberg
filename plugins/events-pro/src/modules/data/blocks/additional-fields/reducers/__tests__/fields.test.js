/**
 * Internal dependencies
 */
import fields from '../fields';
import { actions } from '@moderntribe/events-pro/data/blocks/additional-fields';

describe( 'Fields reducer', () => {
	let state = {};

	beforeEach( () => {
		state = { allIds: [], byId: {} };
	} );

	test( 'Default reducer', () => {
		expect( fields( undefined, {} ) ).toEqual( state );
	} );

	test( 'Add a new block inside of the reducer', () => {
		expect( fields( state, actions.addField( 'Website' ) ) ).toMatchSnapshot();
		expect( fields( state, actions.setFieldName( 'Website' ) ) ).toMatchSnapshot();
	} );

	test( 'Remove an existing block from the reducer', () => {
		state = fields( state, actions.removeField( 'Website' ) );
		expect( fields( state, actions.removeField( 'Website' ) ) ).toMatchSnapshot();
	} );
} );
