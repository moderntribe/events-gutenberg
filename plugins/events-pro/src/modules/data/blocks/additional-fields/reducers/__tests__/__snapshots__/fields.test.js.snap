// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fields reducer Add a new block inside of the reducer 1`] = `
Object {
  "allIds": Array [
    "Website",
  ],
  "byId": Object {
    "Website": Object {
      "dividerEnd": "",
      "dividerList": "",
      "isPristine": true,
      "name": "Website",
      "options": Array [],
      "type": "",
      "value": "",
    },
  },
}
`;

exports[`Fields reducer Add a new block inside of the reducer 2`] = `
Object {
  "allIds": Array [
    "Website",
  ],
  "byId": Object {
    "Website": Object {
      "dividerEnd": "",
      "dividerList": "",
      "isPristine": true,
      "name": "Website",
      "options": Array [],
      "type": "",
      "value": "",
    },
  },
}
`;

exports[`Fields reducer Remove an existing block from the reducer 1`] = `
Object {
  "allIds": Array [],
  "byId": Object {},
}
`;
